operator/cmd/mesh/profile-common.go

```diff
@@ -166,14 +166,26 @@ func genProfile(helmValues bool, inFilename []string, profile, setOverlayYAML, c
 	return finalYAML, err
 }
 
-func unmarshalAndValidateIOP(crYAML string, force bool) (*v1alpha1.IstioOperatorSpec, string, error) {
+func unmarshalAndValidateIOP(crYAML string, force, dumpTranslation bool, l *Logger) (*v1alpha1.IstioOperatorSpec, string, error) {
 	// TODO: add GVK handling as appropriate.
 	if crYAML == "" {
 		return &v1alpha1.IstioOperatorSpec{}, "", nil
 	}
 	iops, _, err := manifest.ParseK8SYAMLToIstioOperatorSpec(crYAML)
 	if err != nil {
-		return nil, "", fmt.Errorf("could not unmarshal the overlay file: %s\n\nOriginal YAML:\n%s", err, crYAML)
+		iopYAML, err := translateICPToIOP(crYAML)
+		if err != nil {
+			return nil, "", fmt.Errorf("could not translate ICP to IOP: %s\n\nOriginal YAML:\n%s", err, crYAML)
+		}
+		iops, _, err = manifest.ParseK8SYAMLToIstioOperatorSpec(iopYAML)
+		if err != nil {
+			return nil, "", fmt.Errorf("could not unmarshal the overlay file: %s\n\nOriginal YAML:\n%s", err, crYAML)
+		}
+		if dumpTranslation {
```

Comment:
i feel like this function is doing too much - i'd recommend a separate helper that returns (string, error) which translates.

Reply:
Fixed.